### PR TITLE
Atlassian Jira and SAP SuccessFactors addons

### PR DESCRIPTION
**Pull request by yavorg**
The Jira one I did for my demo to Iteratively. 

The SuccessFactors one works well, but has the following issues:
- They expect you to generate your own OData client instead of providing a SDK on npm. Unsure why, since you generate it by running against a static OData .edmx file... go figure: https://developers.sap.com/tutorials/cloudsdk-js-generator.html. This results in 20 or so files getting checked into every function - we should really put this on npm
- The auth mechanism uses (API Key) is only good for their sandbox environment. We'd have to switch it to HTTP Basic Username/Password which appears to be what they use for customer instances.

### CODE REVIEW

File: addons/sap-success-factors/install.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+  .sort()
+  .map((k) => `${k}=${storageSpecification[k]}`)
+  .join('\n')}
+`,
+      nodejs: {
+        files: getTemplateFiles([
+          'index.js',
+          'package.json',
+          'success-factors.js',
+          'client/candidate/BatchRequest.js',
```
Comment (bennbollay) [inline addons/sap-success-factors/install.js]: Were these not already published in a package on npm?

File: addons/sap-success-factors/install.js [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+          'client/candidate/JobApplicationSnapshot_TalentPoolcorpRequestBuilder.js',
+          'client/candidate/function-imports.js',
+          'client/candidate/index.js',
+        ]),
+      },
+      metadata: {
+        fusebit: {
+          editor: {
+            navigationPanel: {
+              hideFiles: [
```
Comment (bennbollay) [inline addons/sap-success-factors/install.js]: Create one list and then filter out success-factors.js?

**Review by bennbollay — DISMISSED** [review]
Mechanically looks identical to the existing addon's.

**Comment by bennbollay** [comment]
Notes from watching the demo:
  * Embedded apiKey in the "success-factors.js" should be removed, and natively wrapped under the hood into the SDK.
  * Also the url, that shouldn't need to be explicitly called out when the SDK is used.

File: addons/sap-success-factors/install.js (hunk 1 above)
Comment (yavorg) [reply addons/sap-success-factors/install.js]: Nope, see my comment in the description

Comment (bennbollay) [reply addons/sap-success-factors/install.js]: Well I guess it'll be easier to embed the auth and url aspects into it this way.

Though - is there a reason you're not publishing an NPM package for it?  Speed-to-delivery is a fine reason, just making sure there's not a technical one.

Can you add the comments at the top of this PR into a README or something somewhere in the repo, for the next person to come along and wonder what those files are for?  Or even better, the incantation to generate them, perhaps even as a script in package.json?

File: addons/sap-success-factors/install.js (hunk 2 above)
Comment (yavorg) [reply addons/sap-success-factors/install.js]: Fixed

**Comment by yavorg** [comment]
> Notes from watching the demo:
> 
> * Embedded apiKey in the "success-factors.js" should be removed, and natively wrapped under the hood into the SDK.
> * Also the url, that shouldn't need to be explicitly called out when the SDK is used.

Both the API Key and URL only need to be specified when talking to the sandbox environment. To talk to an actual customer instance, it uses a different auth scheme and it's encapsulated via "destinations" which we can wrap more cleanly. I couldn't find a cleaner way to wrap the API Key approach.

**Pull request merged**